Remotion/Web/Development.WebTesting/WebDriver/Configuration/Chrome/ChromeSecurityWarningsRegistryCleanUpStrategyFactory.cs

```diff
@@ -0,0 +1,128 @@
+{
+  /// <summary>
+  /// Responsible for interacting with the Registry (reading or manipulating the <c>CommandLineFlagSecurityWarningsEnabled</c> entry for Chrome).
+  /// </summary>
+  public class ChromeSecurityWarningsRegistryCleanUpStrategyFactory
+  {
+    private class RestoreSecurityWarningsRegistryCleanUpStrategy : IBrowserSessionCleanUpStrategy
+    {
+      public void CleanUp ()
+      {
+        EnsureWriteAccessToChromePolicyKey();
+
+        using (var key = GetOrCreateChromePoliciesKey())
+        {
+          key.DeleteValue (c_securityWarningsPolicyName);
+        }
+      }
+    }
+
+    private class EmptyCleanUpStrategy : IBrowserSessionCleanUpStrategy
+    {
+      public void CleanUp ()
+      {
+      }
+    }
+
+    private const string c_chromePoliciesPath = @"SOFTWARE\Policies\Google\Chrome";
+    private const string c_securityWarningsPolicyName = "CommandLineFlagSecurityWarningsEnabled";
+
+    private readonly ChromeDisableSecurityWarningsBehavior _disableSecurityWarningsBehavior;
+
+    public ChromeSecurityWarningsRegistryCleanUpStrategyFactory (ChromeDisableSecurityWarningsBehavior disableSecurityWarningsBehavior)
+    {
+      _disableSecurityWarningsBehavior = disableSecurityWarningsBehavior;
+    }
+
+    /// <summary>
+    /// Depending on the set behavior, either throws an exception if <c>CommandLineFlagSecurityWarningsEnabled</c> is not set to false in the registry, attempts to set it,
+    /// or does nothing.
+    /// </summary>
+    /// <returns>
+    /// A <see cref="IBrowserSessionCleanUpStrategy"/>, cleaning up the registry flag if set.
+    /// </returns>
+    /// <exception cref="InvalidOperationException">
+    /// If the behavior <see cref="ChromeDisableSecurityWarningsBehavior.Require"/> is set, and the needed registry key is not set.
+    /// </exception>
+    [NotNull]
+    public IBrowserSessionCleanUpStrategy CreateCleanUpStrategy ()
+    {
+      switch (_disableSecurityWarningsBehavior)
+      {
+        case ChromeDisableSecurityWarningsBehavior.Require when SecurityWarningsEnabled():
+        {
+          throw new InvalidOperationException ($"The '{c_securityWarningsPolicyName}' Chrome policy needs to be set to 0.");
+        }
+        case ChromeDisableSecurityWarningsBehavior.Automatic when SecurityWarningsEnabled():
+        {
+          DisableSecurityWarningsViaRegistry();
+          return new RestoreSecurityWarningsRegistryCleanUpStrategy();
+        }
+        default:
+        {
+          return new EmptyCleanUpStrategy();
+        }
+      }
+    }
+
+    private void DisableSecurityWarningsViaRegistry ()
+    {
+      EnsureWriteAccessToChromePolicyKey();
+
+      using (var key = GetOrCreateChromePoliciesKey())
+      {
+        key.SetValue (c_securityWarningsPolicyName, 0);
+      }
+    }
+
+    private bool SecurityWarningsEnabled ()
+    {
+      using (var localMachineKey = Registry.LocalMachine.OpenSubKey (c_chromePoliciesPath))
```

Comment:
should the order of machne and user be switched to match the evaluation in line 107?